submissions/franchukv/js-oop-frogger/app.js

```diff
@@ -0,0 +1,117 @@
+const CANVAS_HEIGHT = 606;
+const CANVAS_WIDTH = 505;
+const TILE_HEIGHT = 83;
+const TILE_WIDTH = 101;
+const ENEMY_SPEED = 80;
+const PLAYER_STAT = {
+    x: 202,
+    y: 380,
+    height: 40,
+    width: 40,
+};
+
+const Player = function () {
+    this.sprite = 'images/char-boy.png';
+    this.x = PLAYER_STAT.x;
+    this.y = PLAYER_STAT.y;
+    this.height = PLAYER_STAT.height;
+    this.width = PLAYER_STAT.width;
+};
+
+Player.prototype.update = function () {
+    if (this.y < 0) {
+        this.resetPlayerPosition();
+    }
+};
+
+Player.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Player.prototype.resetPlayerPosition = function () {
+    this.x = PLAYER_STAT.x;
+    this.y = PLAYER_STAT.y;
+};
+
+Player.prototype.handleInput = function (key) {
+    switch (key) {
+        case 'up':
+            if (this.y > 0) this.y -= TILE_HEIGHT;
+            break;
+        case 'down':
+            if (this.y <= CANVAS_HEIGHT - TILE_HEIGHT * 3) this.y += TILE_HEIGHT;
+            break;
+        case 'left':
+            if (this.x > 0) this.x -= TILE_WIDTH;
+            break;
+        case 'right':
+            if (this.x < CANVAS_WIDTH - TILE_WIDTH) this.x += TILE_WIDTH;
+            break;
+        default:
+            break;
+    }
+};
+
+const Enemy = function (x, y, player) {
+    this.sprite = 'images/enemy-bug.png';
+    this.x = x;
+    this.y = y;
+    this.width = 80;
+    this.height = 40;
```

Comment:
Looks like enemy and player share some features.

Reply:
Setting constants is yet another way to keep similar things in sync.

P.S. Shared properties and behaviour is a good enough reason to have a base class to encapsulate behaviour and properties handling. We just do not focus on this in this task. Just "a really-really nice to have". In OOP Exercise there are no excuses.

Reply:
Okay, got it, thanks.

Reply:
Please review again, I rewrote the app using constants.